### ISSUE TITLE
Render function fails with exception

### ISSUE THREAD

**Issue by postacik** [chipweinberger/dart_melty_soundfont#11]
Thanks for this great port of MeltySynth  which I use in my C# project.

I'm trying to create a Flutter version of my project but the library has a weird problem which does not exist in the C# version.

Here's the sample application which demonstrates the problem:

[DartMeltySoundFontTest.zip](https://github.com/chipweinberger/DartMeltySoundFont/files/8599711/DartMeltySoundFontTest.zip)

My real application feeds the sound output with synthesized midi data.

The sample application simulates this with the following loop:

![image](https://user-images.githubusercontent.com/3977227/166164163-f380ab29-c2b7-4ec1-a49c-bc56def4d886.png)

However after a few seconds, an exception occurs in the following function: 

![image](https://user-images.githubusercontent.com/3977227/166164042-b3320680-58aa-41b0-9d29-00165ca4e4fa.png)
 
If I encapsulate this part in a try/catch block everything seems to work fine but I'm not sure if it has any side effects or creates performance problems.

Can you please check what's wrong here?

**Comment by chipweinberger** [comment]
Can you screenshot the exception backtrace? and what are the values of the variables `block`, `block.length` and `t`?

**Comment by chipweinberger** [comment]
I just tried your code and, and also tried your SF2 file in my own Flutter Project. No repro. The piano sounded decent! Perhaps your project configuration is somehow causing an issue? Not sure.

Mine:
```
flutter doctor
[✓] Flutter (Channel stable, 2.10.5, on macOS 12.1 21C52 darwin-arm, locale en-US)
[✓] Xcode - develop for iOS and macOS (Xcode 13.2.1)
```

**Comment by postacik** [comment]
Can it be because I'm on Windows?

```
[√] Flutter (Channel stable, 2.10.4, on Microsoft Windows [Version 10.0.19042.1645], locale tr-TR)
[√] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
[√] Chrome - develop for the web
[√] Visual Studio - develop for Windows (Visual Studio Professional 2019 16.11.7)
[√] Android Studio (version 2020.3)
[√] VS Code (version 1.65.1)
```

![image](https://user-images.githubusercontent.com/3977227/166177500-8d907a5b-a05b-4565-9062-5072a950f9e3.png)

![image](https://user-images.githubusercontent.com/3977227/166177567-d3303569-ce97-4014-8a6f-efb9bb71113b.png)

After I added try/catch like this...

![image](https://user-images.githubusercontent.com/3977227/166177890-c1fcc52e-7984-4850-bb2a-485b5eb8e88e.png)

...these are the values which cause exception:

![image](https://user-images.githubusercontent.com/3977227/166177836-b5ccaf34-ba69-4654-bfae-5302cbd1a57a.png)

**Comment by chipweinberger** [comment]
Thanks for the screenshots!

This is actually a bug! I mixed up that List.fillRange wants the *end index* and not a `length`.

**In the original implementation is using:**
definition:  `public static void Clear (Array array, int index, int length);`
code: `Array.Clear(block, t, block.Length - t);`

**Whereas I'm using** 
definition: `void List.fillRange ([int] start,[int] end, [E fillValue])`
code: `block.fillRange(t, block.length - t, 0.0);`

Note that it is *end* and not *length*.

**In other words, the dart code should be:**

`block.fillRange(t, block.length, 0.0);`

I'll push a fix. I have no idea why this has not repro'd on my side though. Maybe I have bounds checking disabled. It's interesting that this bug does not cause any noticeable problems for me. But it is definitely a bug.

**Comment by chipweinberger** [comment]
fixed. try updating. 

 914b72fbe09fe8e6918bf4c37ca4968ecc2db6aa

**Comment by postacik** [comment]
Thanks. It does not raise an exception now.
